File: 2283-sort-even-and-odd-indices-independently/sort-even-and-odd-indices-independently.cpp
class Solution {
public:
    vector<int> sortEvenOdd(vector<int>& nums) {
        vector<int> v;
        int count=0;
        for(int i=0;i<nums.size();i++){
            if(i%2==0){
                v.push_back(nums[i]);
                count++;
            }
        }
        for(int i=0;i<nums.size();i++){
            if(i%2!=0){
                v.push_back(nums[i]);
            }
        }
        auto mid=v.begin()+count;
        sort(v.begin(),mid);
        sort(mid,v.end(),greater<int>());

        int i=0,j=0;
        while(i<nums.size()){
            nums[i]=v[j];
            j++;
            i+=2;
        }
        i=1;
        while(i<nums.size()){
            nums[i]=v[j];
            j++;
            i+=2;
        }
        return nums;
    }
};

File: 2283-sort-even-and-odd-indices-independently/sort-even-and-odd-indices-independently.java
class Solution {
    public int[] sortEvenOdd(int[] nums) {
        int n=nums.length;
        for(int i=0;i<n;i+=2){
            for(int j=i+2;j<n;j+=2){
                if(nums[i]>nums[j]){
                    int temp=nums[i];
                    nums[i]=nums[j];
                    nums[j]=temp;
                }
            }
        }
        for(int i=1;i<n;i+=2){
            for(int j=i+2;j<n;j+=2){
                if(nums[i]<nums[j]){
                    int temp=nums[i];
                    nums[i]=nums[j];
                    nums[j]=temp;
                }
            }
        }
        return nums;
    }
}